File: original_php_project/src/Common/Service/DateLocalizedHelper.php
<?php

namespace App\Common\Service;

use App\Common\Service\Language;
use Illuminate\Support\Facades\Lang;

class DateLocalizedHelper
{
    private const MONTH_NUMBER_TO_NAME_INCLINED = [
        Language::RU => [
            1 => 'Января',
            2 => 'Февраля',
            3 => 'Марта',
            4 => 'Апреля',
            5 => 'Мая',
            6 => 'Июня',
            7 => 'Июля',
            8 => 'Августа',
            9 => 'Сентября',
            10 => 'Октября',
            11 => 'Ноября',
            12 => 'Декабря',
        ],
        Language::UA => [
            1 => 'Січня',
            2 => 'Лютого',
            3 => 'Березня',
            4 => 'Квітня',
            5 => 'Мая',
            6 => 'Червень',
            7 => 'Липня',
            8 => 'Серпня',
            9 => 'Вересня',
            10 => 'Жовтня',
            11 => 'Листопада',
            12 => 'Грудня',
        ],
    ];

    private const WEEK_DAY_NUMBER_TO_NAME = [
        Language::RU => [
            1 => 'Понедельник',
            2 => 'Вторник',
            3 => 'Среда',
            4 => 'Четверг',
            5 => 'Пятница',
            6 => 'Суббота',
            7 => 'Воскресенье',
        ],
        Language::UA => [
            1 => 'Понеділок',
            2 => 'Вівторок',
            3 => 'Середа',
            4 => 'Четвер',
            5 => 'П\'ятниця',
            6 => 'Субота',
            7 => 'Неділя',
        ],
    ];

    private const WEEK_DAY_NUMBER_TO_SHORT_NAME = [
        Language::RU => [
            1 => 'Пн',
            2 => 'Вт',
            3 => 'Ср',
            4 => 'Чт',
            5 => 'Пт',
            6 => 'Сб',
            7 => 'Вс',
        ],
        Language::UA => [
            1 => 'Пн',
            2 => 'Вт',
            3 => 'Ср',
            4 => 'Чт',
            5 => 'Пт',
            6 => 'Сб',
            7 => 'Нд',
        ],
    ];

    public function getMonthNameByNumberInclined(int $month, string $language): string
    {
        return self::MONTH_NUMBER_TO_NAME_INCLINED[$language][$month];
    }

    public function getWeekDayNameByNumber(int $day, string $language): string
    {
        return self::WEEK_DAY_NUMBER_TO_NAME[$language][$day];
    }

    public function getWeekDayShortNameByNumber(int $day, string $language): string
    {
        return self::WEEK_DAY_NUMBER_TO_SHORT_NAME[$language][$day];
    }

    public function getDateTimeGoodLookingLabel(\DateTime $dateTime, string $language = Language::RU): string
    {
        [
            'time' => $time,
            'dayOfWeekFull' => $dayOfWeekFull,
            'dayOfMonth' => $dayOfMonth,
            'month' => $month,
        ] = $this->getDateTimeGoodLookingParts($dateTime, $language);

        return "{$dayOfWeekFull} на {$time}, {$dayOfMonth} {$month}";
    }

    public function getDateTimeGoodLookingParts(\DateTime $dateTime, string $language): array
    {
        $dayOfWeekNumber = $dateTime->format('N');
        $dayOfMonth = $dateTime->format('j');
        $dayOfWeekFull = $this->getWeekDayNameByNumber($dayOfWeekNumber, $language);
        $month = $this->getMonthNameByNumberInclined(
            $dateTime->format('n'),
            $language
        );
        $time = $dateTime->format('H:i');

        return ['time' => $time, 'dayOfWeekFull' => $dayOfWeekFull, 'dayOfMonth' => $dayOfMonth, 'month' => $month, 'year' => $dateTime->format('Y')];
    }
}

File: target_java_project/src/main/java/com/goodhelp/common/service/DateLocalizedHelper.java
package com.goodhelp.common.service;

import org.springframework.context.MessageSource;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.Locale;

/**
 * Helper service for localized date formatting.
 * Migrated from PHP: App\Common\Service\DateLocalizedHelper
 */
@Service
public class DateLocalizedHelper {
    
    private final MessageSource messageSource;
    
    // Week day keys for translation
    private static final String[] WEEKDAY_KEYS = {
        "", // 0 - not used (days start from 1)
        "weekday.monday",
        "weekday.tuesday",
        "weekday.wednesday",
        "weekday.thursday",
        "weekday.friday",
        "weekday.saturday",
        "weekday.sunday"
    };
    
    private static final String[] WEEKDAY_SHORT_KEYS = {
        "",
        "weekday.short.monday",
        "weekday.short.tuesday",
        "weekday.short.wednesday",
        "weekday.short.thursday",
        "weekday.short.friday",
        "weekday.short.saturday",
        "weekday.short.sunday"
    };
    
    private static final String[] MONTH_KEYS = {
        "", // 0 - not used
        "month.january",
        "month.february",
        "month.march",
        "month.april",
        "month.may",
        "month.june",
        "month.july",
        "month.august",
        "month.september",
        "month.october",
        "month.november",
        "month.december"
    };
    
    private static final String[] MONTH_INCLINED_KEYS = {
        "",
        "month.inclined.january",
        "month.inclined.february",
        "month.inclined.march",
        "month.inclined.april",
        "month.inclined.may",
        "month.inclined.june",
        "month.inclined.july",
        "month.inclined.august",
        "month.inclined.september",
        "month.inclined.october",
        "month.inclined.november",
        "month.inclined.december"
    };
    
    public DateLocalizedHelper(MessageSource messageSource) {
        this.messageSource = messageSource;
    }
    
    public String getWeekDayNameByNumber(int dayOfWeek, String localeCode) {
        if (dayOfWeek < 1 || dayOfWeek > 7) {
            throw new IllegalArgumentException("Day of week must be between 1 and 7");
        }
        return messageSource.getMessage(WEEKDAY_KEYS[dayOfWeek], null, getLocale(localeCode));
    }
    
    public String getWeekDayShortNameByNumber(int dayOfWeek, String localeCode) {
        if (dayOfWeek < 1 || dayOfWeek > 7) {
            throw new IllegalArgumentException("Day of week must be between 1 and 7");
        }
        return messageSource.getMessage(WEEKDAY_SHORT_KEYS[dayOfWeek], null, getLocale(localeCode));
    }
    
    public String getMonthNameByNumber(int month, String localeCode) {
        if (month < 1 || month > 12) {
            throw new IllegalArgumentException("Month must be between 1 and 12");
        }
        return messageSource.getMessage(MONTH_KEYS[month], null, getLocale(localeCode));
    }
    
    public String getMonthNameByNumberInclined(int month, String localeCode) {
        if (month < 1 || month > 12) {
            throw new IllegalArgumentException("Month must be between 1 and 12");
        }
        return messageSource.getMessage(MONTH_INCLINED_KEYS[month], null, getLocale(localeCode));
    }
    
    /**
     * Format a datetime in a "good looking" format.
     * Format: "{Weekday} на {time}, {day} {Month}"
     * Example: "Понедельник на 14:30, 15 Января" (Monday at 14:30, 15 January)
     * 
     * @param dateTime the datetime to format
     * @param localeCode locale code (ua, ru, en)
     * @return formatted string
     */
    public String getDateTimeGoodLookingLabel(LocalDateTime dateTime, String localeCode) {
        DateTimeGoodLookingParts parts = getDateTimeGoodLookingParts(dateTime, localeCode);
        // Format: "{Weekday} на {time}, {day} {Month}"
        return String.format("%s на %s, %d %s",
            parts.dayOfWeekFull(),
            parts.time(),
            parts.dayOfMonth(),
            parts.month()
        );
    }
    
    /**
     * Get parts of a datetime formatted for display.
     * 
     * @param dateTime the datetime to format
     * @param localeCode locale code (ua, ru, en)
     * @return parts object with time, dayOfWeekFull, dayOfMonth, month, year
     */
    public DateTimeGoodLookingParts getDateTimeGoodLookingParts(LocalDateTime dateTime, String localeCode) {
        int dayOfWeekNumber = dateTime.getDayOfWeek().getValue(); // 1 = Monday, 7 = Sunday
        int dayOfMonth = dateTime.getDayOfMonth();
        String dayOfWeekFull = getWeekDayNameByNumber(dayOfWeekNumber, localeCode);
        String month = getMonthNameByNumberInclined(dateTime.getMonthValue(), localeCode);
        String time = dateTime.format(DateTimeFormatter.ofPattern("HH:mm"));
        int year = dateTime.getYear();
        
        return new DateTimeGoodLookingParts(time, dayOfWeekFull, dayOfMonth, month, year);
    }
    
    /**
     * Record for datetime parts.
     */
    public record DateTimeGoodLookingParts(
        String time,
        String dayOfWeekFull,
        int dayOfMonth,
        String month,
        int year
    ) {}
    
    private Locale getLocale(String localeCode) {
        if (localeCode == null || localeCode.isEmpty()) {
            return Locale.ENGLISH;
        }
        // Map "ua" to "uk" (Ukrainian language code)
        if ("ua".equalsIgnoreCase(localeCode)) {
            localeCode = "uk";
        }
        return Locale.forLanguageTag(localeCode);
    }
}
